server/src/test/java/org/solarbank/server/unit/dto/CalculateRequestTest.java

```diff
@@ -0,0 +1,115 @@
+package org.solarbank.server.unit.dto;
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.solarbank.server.dto.CalculateRequest;
+import org.solarbank.server.dto.EnergyTariff;
+import org.solarbank.server.dto.Location;
+import org.solarbank.server.dto.PanelSize;
+import org.solarbank.server.ValidationMessage;
+
+import java.util.Set;
+
+import static org.solarbank.server.CreateMockCalculateRequest.createCalculateRequest;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CalculateRequestTest {
+    private Validator validator;
+
+    private CalculateRequest calculateRequest = createCalculateRequest();
+
+    @BeforeEach
+    public void setUp() {
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
```

Comment:
This should be called inside a try with resources block as it implements `AutoCloseable` e.g.

```
        try (ValidatorFactory factory = Validation.buildDefaultValidatorFactory()) {
            validator = factory.getValidator();
        }
```

When you wrap it like this it calls `.close()` to prevent a memory leak.

Reply:
changed this